structuralcodes/sections/_generic.py

```diff
@@ -0,0 +1,461 @@
+""""Generic class section implemenetation"""
+from __future__ import annotations
+
+import typing as t
+import warnings
+from numpy.typing import ArrayLike
+import numpy as np
+from shapely.geometry import (Polygon,
+                              Point,
+                              LineString,
+                              LinearRing,
+                              MultiPolygon,
+                              MultiLineString)
+from shapely.ops import split
+from structuralcodes.core.base import Material
+from structuralcodes.core.base import Section, SectionCalculator
+import structuralcodes.sections._section_results as s_res
+
+# Useful classes and functions: where to put?????? (core?
+# utility folder in sections? here in this file?)
+# Polygons, LineStrings, Points, MultiLyneStrings, MultiPolygons etc.
+
+# to think: dataclass or class?
+# Note that some things are already computed (like area) by shapely
+# like: polygon.area, polygon.centroid, etc.
+
+# For now dataclass, if we need convert to regular class,
+# init commented for now
+
+
+class Geometry:
+    '''Base class for a geometry object'''
```

Comment:
Not crucial, but in other places we tend to use double quotes for the docstrings and single ones for inline strings.